### ISSUE TITLE
Workflow draw_args not applied

### ISSUE THREAD

**Issue by Roang-zero1** [caksoylar/keymap-drawer#45]
I am trying to use the ZMK generation as part of my GitHub workflow, unfortunately the `draw_args` are not applied.

As you can see in [my config](https://github.com/Roang-zero1/zmk-soflev2/blob/main/.github/workflows/draw-keymaps.yml#L21) I want to select the keybaord with `--qmk-keybard` or `-k`.
When I do this on the CLI it works as intended, when I try to apply this to the action it unfortunately does not apply the keybard. (latest log: https://github.com/Roang-zero1/zmk-soflev2/actions/runs/6095789469/job/16540062027)

PS: The sample workflow is missing the permissions to write back to the repository: e.g. https://github.com/Roang-zero1/zmk-soflev2/blob/main/.github/workflows/draw-keymaps.yml#L13-L14

**Comment by caksoylar** [comment]
The draw args is a mapping from keyboard name to the arguments, so try the following instead:
```yaml
draw_args: "splitkb_aurora_sofle:'--qmk-keyboard splitkb/aurora/sofle_v2/rev1'"
```

That being said, I [just added support for it](https://github.com/caksoylar/keymap-drawer/commit/39f7d7a16530f74c6e4b72d4136a86a8219d6117) in the default mapping so hopefully it should work without custom `draw_args` if you use the main branch via `install_branch: main` in the workflow inputs.

> PS: The sample workflow is missing the permissions to write back to the repository: e.g. https://github.com/Roang-zero1/zmk-soflev2/blob/main/.github/workflows/draw-keymaps.yml#L13-L14

I added it to the reusable workflow itself via https://github.com/caksoylar/keymap-drawer/blob/main/.github/workflows/draw-zmk.yml#L51, but maybe it doesn't work that way. Thanks, I'll have a look.

**Comment by Roang-zero1** [comment]
Thanks for the quick response, I can confirm both variants work.

I think the permissions need to be given in the workflow configuration of the affected repository...

**Comment by Roang-zero1** [comment]
For prosperity: If you want to include the generated image in the Readme, you can do it via `![Current Keybpard Layout](keymap-drawer/<keyboard>.svg?raw=true "Current Keybpard Layout")`

**Comment by caksoylar** [comment]
Good trick with the `?raw=true` 👍

**Comment by debashisbiswas** [comment]
For permissions, I had to make the same change as well to get the workflow to build: 
https://github.com/debashisbiswas/Adv360-Pro-ZMK/blob/V3.0/.github/workflows/draw-keymaps.yml#L14-L15

Seems like this needs to be specified in the repo that is using the workflow for security (for example I have to explicitly allow you to make commits in my repo, you can't allow yourself)
https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

**Comment by caksoylar** [comment]
Thanks @debashisbiswas for confirming the issue (and reminding me to fix it)! [a6e3b18](https://github.com/caksoylar/keymap-drawer/commit/a6e3b188a8417c8c425e9a9e8a2a8373b3e21bc8)
